README.md

```diff
@@ -45,3 +45,21 @@ The following steps should serve as a guide when adding new packages or dependen
 - Add a patch if necessary to `patches` directory.
 - If it's a system dependency, add a build rule to `install.sh`. According to the library, it may be necessary to use `build_library` script or adding a special rule to `build_library_with_toolchain`.
 - That's it! Test your build and watch out build errors. Using `verbose` flags when building should help finding hints about the problems that may arise.
+
+## Building your own apps and libraries on top of ROS cross-compiled workspace
+
+You can use `build_catkin_workspace` as a standalone script to build your Android application or library as a catkin package,
+making use of the ROS cross compiled libraries and system dependencies. The examples under [example_workspace](https://github.com/Intermodalics/ros_android/tree/kinetic/example_workspace/src) can be used as a starting point and as a reference to build your code; in short they are Gradle projects wrapped as Catkin packages.
+
+Here's some guidelines to take into account when creating new projects, using `hello_world` app as an example:
+- Your code should have the structure of a ROS Android package, with a top level `CMakeLists` to make Gradle interact with Catkin. Take [this CMakeLists file](https://github.com/Intermodalics/ros_android/blob/kinetic/example_workspace/src/hello_world_example_app/CMakeLists.txt) as an example.
+- A [package.xml](https://github.com/Intermodalics/ros_android/blob/kinetic/example_workspace/src/hello_world_example_app/package.xml) has to be created, declaring Catkin as the build tool, and the ROS dependencies.
+- In the [bottom level gradle buildscript](https://github.com/Intermodalics/ros_android/blob/kinetic/example_workspace/src/hello_world_example_app/app/build.gradle), add the necessary arguments to the `externalNativeBuild` block, and specify ABI filters as in [here](https://github.com/Intermodalics/ros_android/blob/kinetic/example_workspace/src/hello_world_example_app/app/build.gradle#L10-L23). Then, specify the path to the bottom level CMakeLists file as in [here](https://github.com/Intermodalics/ros_android/blob/kinetic/example_workspace/src/hello_world_example_app/app/build.gradle#L31-L36).
```

Comment:
Mention that the current approach does not allow to build multi-target apks and the architecture/ABI is fixed in [config.sh](https://github.com/Intermodalics/ros_android/blob/17b91ac70fed183f5bf55ed475292c3437d413ef/scripts/config.sh#L5)?

Reply:
Addressed in https://github.com/Intermodalics/ros_android/pull/111/commits/24e38e17adb9dbc86b32fa07736ea485c4bfcc7e.